Source/automap.cpp

```diff
@@ -869,6 +873,10 @@ void AutomapZoomOut()
 
 void DrawAutomap(const Surface &out)
 {
+	if (!*sgOptions.Graphics.showTownMap && leveltype == DTYPE_TOWN) {
+		DrawAutomapText(out);
+		return;
+	}

```

Comment:
Maybe we could instead split this function up in one that draws the text and one that draws the map and then have the one that draws the map do an early return if disabled. That way we don't duplicate code and make it harder to follow.